model/host/host.go

```diff
@@ -3136,6 +3136,17 @@ func CountSpawnhostsWithNoExpirationByUser(ctx context.Context, user string) (in
 	return Count(ctx, query)
 }
 
+// CountIntentHosts counts the number of intent hosts Evergreen will soon
+// attempt to create
+func CountIntentHosts(ctx context.Context) (int, error) {
+	query := bson.M{
+		IdKey:        bson.M{"$regex": "^evg.*"},
```

Comment:
Just out of curiosity, what does this regex check for/mean business wise? Do we keep our hosts around in the db forever and would that cause this query to be slow enough to affect the host allocator job?

Reply:
And would it be worth extracting it to a constant or commenting / referencing where it comes from?

Reply:
The regex checks that the _id starts with "evg-" which is how we determine if a host is an intent hosts. I think it's simplest to keep its usage inline here.

Hosts are kept forever but the hosts collection has an index on the `status` field, so counting the number of hosts that are up should always be efficient since almost all hosts in the collection are terminated.